h2o-core/src/test/java/water/rapids/RapidsTest.java

```diff
@@ -688,13 +688,20 @@ static boolean checkSaneFrame_impl() {
   }
 
   @Test public void test_frameKeyStartsWithNumber() {
-    Frame fr = parse_test_file(Key.make("123STARTSWITHDIGITS"), "smalldata/logreg/prostate.csv");
+    Frame fr = null;
+    Key key = null;
+    
     try {
+      Scope.enter();
+      key = Key.make("123STARTSWITHDIGITS");
```

Comment:
Moving the parsing in the try-finally doesn't prevent a key leakage. If parse_test_file fails you won't have the Frame reference (fr) to remove the Frame. The previous code was doing the same thing.

Reply:
If e.g. Key.make() fails in the previous version, then the finally blocks fails on NullPointerException, that was the main drive for this "fix"

Reply:
If `Key.make()` fails the finally block fails on NPE but no key is leaked. Whole H2O depends on Key.make so handling NPE in that case is IMHO not necessary.